tikv/server.go

```diff
@@ -562,6 +565,44 @@ func (svr *Server) UnsafeDestroyRange(context.Context, *kvrpcpb.UnsafeDestroyRan
 	return &kvrpcpb.UnsafeDestroyRangeResponse{}, nil
 }
 
+// deadlock detection related services
+// GetWaitForEntries tries to get the waitFor entries
+func (svr *Server) GetWaitForEntries(ctx context.Context,
+	req *deadlockPb.WaitForEntriesRequest) (*deadlockPb.WaitForEntriesResponse, error) {
+	// TODO
+	return &deadlockPb.WaitForEntriesResponse{}, nil
+}
+
+// Detect will handle detection rpc from other nodes
+func (svr *Server) Detect(stream deadlockPb.Deadlock_DetectServer) error {
+	for {
+		req, err := stream.Recv()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+		if svr.mvccStore.deadlockDetector.isLeader() {
```

Comment:
We should break the loop if it's the leader.

Reply:
Break the loop if it's not the leader? @coocood

Reply:
yes, my typo